### PR TITLE
docs: translate understanding your ui as a tree

### LINKED ISSUE #659: Understanding Your UI as a Tree

### PR DESCRIPTION
**Pull request by r17x**
Closes #659 

## Description

Translate the  page.
Page URL:

### CODE REVIEW

**Comment by resir014** [comment]
@r17x Don't forget to also update the corresponding sidebar in `sidebarLearn.json`:
https://github.com/reactjs/id.react.dev/blob/main/src/sidebarLearn.json#L90-L93

**Comment by r17x** [comment]
Thank you for remind me, I will update @resir014

File: src/content/learn/understanding-your-ui-as-a-tree.md [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
+<Diagram name="preserving_state_dom_tree" height={193} width={864} alt="Diagram dengan tiga bagian yang disusun secara horizontal. Pada bagian pertama, terdapat tiga persegi panjang yang ditumpuk secara vertikal, dengan label 'Komponen A', 'Komponen B', dan 'Komponen C'. Transisi ke panel berikutnya adalah sebuah panah dengan logo React di bagian atas yang berlabel 'React'. Bagian tengah berisi sebuah pohon komponen, dengan root berlabel 'A' dan dua komponen berlabel 'B' dan 'C'. Bagian selanjutnya ditransisikan lagi menggunakan panah dengan logo React di bagian atas berlabel 'React DOM'. Bagian ketiga dan terakhir adalah wireframe dari sebuah peramban, yang berisi sebuah pohon dengan 8 *node*, yang hanya memiliki sebuah bagian yang disorot (mengindikasikan subpohon dari bagian tengah).">
```
Comment (zainfathoni) [inline src/content/learn/understanding-your-ui-as-a-tree.md:25]:
```suggestion
<Diagram name="preserving_state_dom_tree" height={193} width={864} alt="Diagram dengan tiga bagian yang disusun secara horizontal. Pada bagian pertama, terdapat tiga persegi panjang yang ditumpuk secara vertikal, dengan label 'Komponen A', 'Komponen B', dan 'Komponen C'. Transisi ke panel berikutnya adalah sebuah panah dengan logo React di bagian atas yang berlabel 'React'. Bagian tengah berisi sebuah pohon komponen, dengan akar berlabel 'A' dan dua komponen anak berlabel 'B' dan 'C'. Bagian selanjutnya ditransisikan lagi menggunakan panah dengan logo React di bagian atas berlabel 'React DOM'. Bagian ketiga dan terakhir adalah *wireframe* dari sebuah peramban, yang berisi sebuah pohon dengan 8 *node*, yang hanya memiliki sebuah bagian yang disorot (mengindikasikan subpohon dari bagian tengah).">
```

File: src/content/learn/understanding-your-ui-as-a-tree.md [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
+Seperti halnya peramban (*browser*) dan perangkat seluler (*mobile*) bergerak, React juga menggunakan struktur pohon untuk mengelola dan memodelkan hubungan antar komponen dalam aplikasi React. Pohon-pohon ini adalah alat yang berguna untuk memahami bagaimana data mengalir melalui aplikasi React dan bagaimana mengoptimalkan *rendering* dan ukuran aplikasi.
 
-## The Render Tree {/*the-render-tree*/}
+## Pohon *Render* {/*the-render-tree*/}
 
-A major feature of components is the ability to compose components of other components. As we [nest components](/learn/your-first-component#nesting-and-organizing-components), we have the concept of parent and child components, where each parent component may itself be a child of another component.
+Fitur utama dari komponen adalah kemampuan untuk menyusun komponen dari komponen lain. Saat kita [menyusun komponen](/learn/
+your-first-component#nesting-and-organizing-components)
+, kita memiliki konsep komponen induk dan anak, di mana setiap komponen induk dapat menjadi anak dari komponen lain.
```
Comment (zainfathoni) [inline src/content/learn/understanding-your-ui-as-a-tree.md:36]:
```suggestion
Fitur utama dari komponen adalah kemampuan untuk menyusun komponen dari komponen lain. Saat kita [menyusun komponen](/learn/your-first-component#nesting-and-organizing-components), kita memiliki konsep komponen induk dan anak, di mana setiap komponen induk dapat menjadi anak dari komponen lain.
```
Keep the link intact.

**Review by zainfathoni — CHANGES_REQUESTED** [review]

File: src/content/learn/understanding-your-ui-as-a-tree.md [hunk 3]
```diff
@@ -118,34 +118,36 @@ export default [
 
 </Sandpack>
 
-<Diagram name="render_tree" height={250} width={500} alt="Tree graph with five nodes. Each node represents a component. The root of the tree is App, with two arrows extending from it to 'InspirationGenerator' and 'FancyText'. The arrows are labelled with the word 'renders'. 'InspirationGenerator' node also has two arrows pointing to nodes 'FancyText' and 'Copyright'.">
+<Diagram name="render_tree" height={250} width={500} alt="Graf pohon dengan lima simpul. Tiap simpul merepresentasikan sebuah komponen. Akar dari pohon ini adalah App, dengan dua anak panah yang memanjang dari akar tersebut ke 'InspirationGenerator' dan 'FancyText'. Panah-panah tersebut diberi label dengan kata 'render'. Node 'InspirationGenerator' juga memiliki dua anak panah yang mengarah ke node 'FancyText' dan 'Copyright'.
+">
```
Comment (zainfathoni) [inline src/content/learn/understanding-your-ui-as-a-tree.md:122]:
```suggestion
<Diagram name="render_tree" height={250} width={500} alt="Graf pohon dengan lima simpul. Tiap simpul merepresentasikan sebuah komponen. Akar dari pohon ini adalah App, dengan dua anak panah yang memanjang dari akar tersebut ke 'InspirationGenerator' dan 'FancyText'. Panah-panah tersebut diberi label dengan kata 'render'. Node 'InspirationGenerator' juga memiliki dua anak panah yang mengarah ke node 'FancyText' dan 'Copyright'.">
```

File: src/content/learn/understanding-your-ui-as-a-tree.md [hunk 4]
```diff
@@ -118,34 +118,36 @@ export default [
...
+Pohon ini terdiri dari beberapa simpul, yang masing-masing mewakili sebuah komponen. `App`, `FancyText`, `Copyright`, dan beberapa lainnya, merupakan beberapa simpul di dalam pohon kita.
 
-The root node in a React render tree is the [root component](/learn/importing-and-exporting-components#the-root-component-file) of the app. In this case, the root component is `App` and it is the first component React renders. Each arrow in the tree points from a parent component to a child component.
+Simpul akar (*root node*) dalam pohon *render* React adalah [komponen akar](/learn/importing-and-exporting-components#the-root-component-file) dari aplikasi. Dalam kasus ini, Akar komponen adalah `Aplikasi` dan merupakan komponen pertama yang di-*render* oleh React. Setiap panah pada pohon menunjuk dari komponen induk ke komponen anak.
 
 <DeepDive>
 
-#### Where are the HTML tags in the render tree? {/*where-are-the-html-elements-in-the-render-tree*/}
+#### Di mana letak dari tag HTML dalam pohon *render*? {/*di-mana-letak-dari-tag-html-dalam-pohon-render*/}
+{/*where-are-the-html-elements-in-the-render-tree*/}
```
Comment (zainfathoni) [inline src/content/learn/understanding-your-ui-as-a-tree.md:138]:
```suggestion
#### Di mana letak dari tag HTML dalam pohon *render*? {/*where-are-the-html-elements-in-the-render-tree*/}
```

**Review by zainfathoni — CHANGES_REQUESTED** [review]

**Pull request merged**